### ISSUE TITLE
Annotation suggestion endpoint returns `qualifier` instead of `property` block

### ISSUE THREAD

**Issue by g1eb** [usc-isi-i2/t2wml#439]
While annotating the `annotation_example.xlsx` file during the demo today we noticed that the suggestion api returns `qualifier` as the suggestion for what should be a `property` in column B, see related file attached here:

[annotation_example.xlsx](https://github.com/usc-isi-i2/t2wml/files/6347175/annotation_example.xlsx)

**Comment by devowit** [comment]
the annotation suggester doesn't always get the right answer, it just makes a best guess based on heuristics. that's why it also returns other relevant options as well. right now property is only suggested as first option for single cell blocks, multi cell blocks are guessed as qualifier. i can add additional refinements like, if the cell has already been wikified to a property it's a property, but no matter what we expect the suggestions to occasionally be wrong...

**Comment by g1eb** [comment]
We need to create a document outlining the inner workings of the algorithm used to suggest the correct annotations, including the different types of content and how it's matched to a role/type/etc.

We can then use this document to discuss how to improve the various block suggestions.

@devowit can you take a first pass on this document please?

**Comment by g1eb** [comment]
I have a suggestion that could resolve ambiguity between qualifier and property blocks spanning multiple cells.

Annotating the file step by step the user needs to select a dependent variable, main subject and property. The guide is asking users to select the variable first, main subject second and property last (we decided not to require/flash missing qualifier or unit annotations).

In other words if the user annotated a block as dependent variable, and another block as the main subject, chances are the next annotation would be the property, maybe that information can be used as a tie breaker between qualifier and property suggestions?

If the user selects a block of years as their third block we could use the property P585 as additional information to suggest qualifier instead of property (even though we expected the third annotation to be a property).

**Comment by szeke** [comment]
Makes sense, once we have subject/property/variable, the rest are likely to be qualifiers or units. We may be able to recognize units using a dictionary of common units, mined from Wikidata

**Comment by g1eb** [comment]
yeah, what do you think about training a simple classifier specific to the causx type of data, e.g. country -> main subject, years -> qualifier, etc. together with the existing heuristics should be possible to improve what we have so far.
